presto-main/src/main/java/com/facebook/presto/execution/TaskExecutor.java

```diff
@@ -855,6 +888,20 @@ public long getRunningTasksLevel4()
     }
 
     @Managed
+    public long getLongestSplitTimeMillis()
+    {
+        Iterator<SplitProfile> iterator = splitProfiler.iterator();
```

Comment:
You should be able to just call `first()` on the `SortedSet` (and handle `NoSuchElementException`).